issuing_using_did.md

```diff
@@ -0,0 +1,400 @@
+              "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6#owner",
+              "type": "Secp256k1VerificationKey2018",
+              "ethereumAddress": "0xe6fe788d8ca214a080b0f6ac7f48480b2aefa9a6",
+              "owner": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6"
+            }
+          },
+          {
+            "type": "DNS-DID",
+            "location": "example.com"
+          }
+        ]
+      }
+    ]
+  },
+  "privacy": {
+    "obfuscatedData": []
+  },
+  "signature": {
+    "type": "SHA3MerkleProof",
+    "targetHash": "61dc9186345e05cc2ae53dc72af880a3b66e2fa7983feaa6254d1518540de50a",
+    "proof": [],
+    "merkleRoot": "61dc9186345e05cc2ae53dc72af880a3b66e2fa7983feaa6254d1518540de50a"
+  }
+}
+```
+
+Issued via Document Store, identified via DID
+
+```json
+{
+  "schema": "tradetrust/v1.0",
+  "data": {
+    "issuers": [
+      {
+        "name": "TradeTrust Demo",
+        "documentStore": "0x6d71da10Ae0e5B73d0565E2De46741231Eb247C7",
+        "identityProof": [
+          {
+            "type": "DID",
+            "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6",
+            "purpose": "CONTRACT_ADDRESS_PROOF",
+            "signature": "",
+            "key": {
+              "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6#owner",
+              "type": "Secp256k1VerificationKey2018",
+              "ethereumAddress": "0xe6fe788d8ca214a080b0f6ac7f48480b2aefa9a6",
+              "owner": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6"
+            }
+          }
+        ]
+      }
+    ]
+  },
+  "privacy": {
+    "obfuscatedData": []
+  },
+  "signature": {
+    "type": "SHA3MerkleProof",
+    "targetHash": "61dc9186345e05cc2ae53dc72af880a3b66e2fa7983feaa6254d1518540de50a",
+    "proof": [],
+    "merkleRoot": "61dc9186345e05cc2ae53dc72af880a3b66e2fa7983feaa6254d1518540de50a"
+  }
+}
+```
+
+Issued by direct signing, identified by DID
+
+```json
+{
+  "schema": "tradetrust/v1.0",
+  "data": {
+    "issuers": [
+      {
+        "name": "TradeTrust Demo",
+        "identityProof": [
+          {
+            "type": "DID",
+            "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6",
+            "purpose": "DOCUMENT_ISSUANCE",
+            "revocation": {
+              "type": "REVOCATION_STORE",
+              "address": "0xabcd...1234"
+            },
+            "key": {
+              "id": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6#owner",
+              "type": "Secp256k1VerificationKey2018",
+              "ethereumAddress": "0xe6fe788d8ca214a080b0f6ac7f48480b2aefa9a6",
+              "owner": "did:ethr:0xE6Fe788d8ca214A080b0f6aC7F48480b2AEfa9a6"
+            }
+          }
+        ]
+      }
+    ]
+  },
+  "proof": {
+    "type": "DidGenericSignature",
+    "proofPurpose": "assertionMethod",
+    "signature": "<signed merkle root>"
+  }
+}
+```
+
+## Pseudocode for verification
+
+### Identity verification Fragments
```

Comment:
What about you have multiple identity proof (for instance DID and DID-DNS) and one is misconfigured (so in that case DID-DNS), what will be the overall status ?

And more generally what happen when you have X identity proof, and one is failing.

Reply:
Thinking about the current imeplementation:

INVALID + VALID = INVALID

*Any invalid, the entire section fails

Reply:
but is it correct ? You still can resolve that did is valid, just dns is not. I think we had that discussion a while ago. why if you have multiple identity verifiers, because one is not able to resolve then it will invalidate all other resolution ?

So in that example, did is valid, dns is not, why dns invalidate did ?

Reply:
i think it is correct, if you put the identifier there then it better be resolving. It's not skipping but its failing...

Reply:
it's skipping on the dns resolution, not the did resolution, id did is an identifier by itself, then the resolution of the did identity is valid even if the resolution of the dns is invalid

Reply:
Yea but overall I see it as...

Raymond is declaring that he is Raymond and Laurent
The check for Raymond pass but the check on Laurent's declaration fail

I think we should fail the entire section in this case since there is illegitimate claims.

Reply:
It's the same logic as when we decided to fail the entire section if any test fails...

Reply:
I dont fully agree, it's kind of the same problem that we currently have for instance in opencerts. OpenCerts has 2 identity verifiers:
- dns
- registry

DNS is skipped of course if there is no DNS entries. But registry check can't be skipped. That means if I apply the same logic if dns is valid and registry is not (i.e often) then the identity is failed. (pls dont argue that OC is a deviant child and I'm not saying we must support oc use case :) )

(Read more on telegram the 20/05/2020)

What I want to point to is that with those mechanism we may have 2 different identities (or more ?) to verify (DID and DNS for instance if we consider DID and DNS being top level identity identifiers):
- why would the failure of one invalidate the other while when they are separate everything is fine (even considering there is a network problem and not a configuration problem)

This also makes me raise another question, why not making dns identity mandatory even with did as long as there are not stronger identify verifier.

Note: I think it's better than we discuss directly rather than arguing over github :p. I wont argue more here, jsut gave my thoughts

Reply:
skip the check if its not found?

Reply:
opencerts was enforced at the verifier level in this case. The certificate issuer did not explicitly claimed that they are under the registry.

Reply:
moving forward you want to be explicit about the claims and the proofs. In the case of opencerts, there was never a claim in the first place. As a result you cannot fail them because they did not claim an identity in the registry in the first place.

Reply:
Maybe OA can follow the proposed pattern (VALID + INVALID + INVALID) and up to consumer to choose a different path